### ISSUE TITLE
GNOME/Wayland: "Huge" menu hiding unexpectedly when moving a mouse

### ISSUE THREAD

**Issue by polter-rnd** [fltk/fltk#1115]
**Describe the bug**
On GNOME+Wayland, "Huge" menu from `menubar` test behaves weirdly sometimes. When moving a mouse down to unveil more menu items, the menu bar hides unexpectedly. The videos are below.

With GNOME fractional scaling enabled (150%) it is very easy reproducible:

https://github.com/user-attachments/assets/ae593d05-e66f-4b3f-bb81-c405ca56cb6c

Without fractional scaling at normal 100% it is more tricky to reproduce but still possible:

https://github.com/user-attachments/assets/7d1d0895-2474-4e5e-9b50-add5664364d6

The menu hides like I'm clicking somewhere outside, but actually I'm not clicking anywhere, just hovering.

**To Reproduce**

Launch `menubar` test and play with "Huge" menu as shown on videos.

**Expected behavior**
Menu does not hide unexpectedly

**FLTK Version**
 - Downloaded and self-built from latest master (`release-1.4.0rc3`)

**Operating System / Platform:**
Please be as precise as possible, e.g. "Linux: Ubuntu 20.04"
 - OS: Linux, GNOME/Wayland
 - OS Version: Fedora 39

**Linux/Unix Runtime, if applicable:**
 - Wayland

**Comment by ManoloFLTK** [comment]
@polter-rnd Does this bug affect an application of yours? or is it only a curiousity revealed by the arguably unrealistic huge menu of the menubar app?

**Comment by polter-rnd** [comment]
No, in app with normal menu size (if it doesn't overflow with scrolling) this issue does not reproduce. I just noticed this issue in `menubar` test

**Comment by MatthiasWM** [comment]
I can't replicate this at the moment on my emulated machine, but that doesn't mean much. I will see if I find the time tomorrow to drill a bit deeper.

**Comment by MatthiasWM** [comment]
@ManoloFLTK It seems to me that Fl_Wayland_Screen_diver::output::wld_scale needs can't be an integer. Ubuntu offers fractional scaling of the desktop (which is marked experimental, but reachable by the user in 22.04. In 24.04 it seems to be blocked by default but can be enabled with `gsettings set org.gnome.mutter experimental-features "['scale-monitor-framebuffer']"`). If scaling is not 100%, 200%, or 300%, the calculations for `Fl::h()` (and others) are off, which in the end causes the menu window to close because the mouse position becomes implausible.

I will take a close look at wld_scale. I assume it would need to be a floating point variable that would be set to 1.25 or 1.5 etc. for fractional screen scaling.

**Comment by MatthiasWM** [comment]
PS: Oh! I had no idea what a wasp's nest the whole fractional scaling on Wayland seems to be. Anyway, here is one resource on how to query fractional scaling from FLTK: https://wayland.app/protocols/fractional-scale-v1 .

**Comment by ManoloFLTK** [comment]
@MatthiasWM I believe FLTK handles fractional scaling correctly. Don't spend time on this before seeing what's in FLTK already. More later.

**Comment by ManoloFLTK** [comment]
@MatthiasWM Please read this first: https://www.fltk.org/doc-1.4/wayland-devel.html#wayland-display

**Comment by ManoloFLTK** [comment]
@MatthiasWM Overall, my conclusion, and what I observed with Mutter and KDE, is that fractional scaling as set under gnome or KDE does this:
- it sends to any Wayland client app (FLTK apps included of course) an integer-valued scale factor (called `wld_scale` under FLTK), 
- it receives from its client apps buffers whose dimension correspond to this integer-valued scale factor, 
- and it scales down at the compositing stage such buffers to present a fractionally scaled result to the display viewer.

**Comment by polter-rnd** [comment]
@MatthiasWM @ManoloFLTK Thanks for your attention! I suspect that fractional scaling is not the root cause here since I was able to catch something similar even without it, but it's the simplest way to reproduce the issue

**Comment by ManoloFLTK** [comment]
@polter-rnd Commit 9b5d40a should fix this issue. Can you confirm?

**Comment by polter-rnd** [comment]
Thanks, I'm no more able to reproduce it on latest master. Confirming
